MSMAll/scripts/SingleSubjectConcat.sh

```diff
@@ -163,31 +141,41 @@ for ((index = 0; index < ${#fMRINamesArray[@]}; ++index)) ; do
 	FrameOutput=${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_${FrameString}${OutputProcSTRING}.dtseries.nii
 
 	# pick frames
-	${Caret7_Command} -cifti-merge ${FrameDenseTCS} -cifti ${FullDenseTCS} -column ${StartFrame} -up-to ${EndFrame}
+	if [[ "$EndFrame" == "" ]]; then
+		if ((StartFrame==1)); then
+			# override the FrameDenseTCS with the full dense timeseries
+			FrameDenseTCS=${FullDenseTCS}
+			# mark temp files for mean, and timeseries after the above process
+			tempfiles_add ${FrameMean} ${FrameOutput}
```

Comment:
It is fine to `tempfiles_add` something that won't exist, this can be moved out of the conditionals for simplicity.

Reply:
I don't add `tempfiles_add {FrameDenseTCS}` because in this particular case, I don't want the full dense timeseries to be deleted...

Reply:
Right, but if you do it before the conditional on the temporary file that might not get created, it is fine.  In fact, you should generally `tempfiles_add` files before you actually create them, so that they don't get left behind if the script errors or is interrupted.

Reply:
I see. Thanks for the explanation.